### PR TITLE
[Ubuntu] Use same kubeconfig.yaml as Amazon Linux

### LINKED ISSUE #355: Check if Ubuntu supports kubelet config file

### PR DESCRIPTION
**Pull request by davidhao3300**
### Description

Switch flag usage in Ubuntu node bootstrapping to usage of kubelet.yaml, similar to how Amazon Linux does it.

Discovered this discrepancy after encountering an issue similar to https://github.com/weaveworks/eksctl/issues/350, and examining the PR addressing that issue showed that the fix was only done for Amazon Linux, but not Ubuntu.

### Checklist

- [x] Code compiles correctly (i.e `make build`)
- [x] Added tests that cover your change (if possible)
- [x] All unit tests passing (i.e. `make test`)
- [ ] Added/modified documentation as required (such as the `README.md`, and `examples` directory)
- [x] Manually tested

closes #355

### CODE REVIEW

**Comment by martina-if** [comment]
I tested it for `ami-0c5febddf2149f5f4` and it seems to work fine :+1:

**Comment by davidhao3300** [comment]
Is there anything else needed to get the PR merged? First time contributing a change to this repo

**Comment by martina-if** [comment]
Hi @davidhao3300 I would like someone else in my team to take a look, since I don't have all the context on why this was done in this way compared to the AL2 images bootstrapping. Hopefully we will merge this today :)

**Comment by davidhao3300** [comment]
Makes sense, thanks!

**Comment by errordeveloper** [comment]
This will close https://github.com/weaveworks/eksctl/issues/355.

**Comment by davidhao3300** [comment]
Updated!

**Comment by davidhao3300** [comment]
anyone mind approving/merging? want to get rid of my forked binary lol

File: Makefile [hunk 1]
```diff
@@ -135,7 +135,7 @@ check-generated-sources-up-to-date: regenerate-sources
 pkg/addons/default/assets.go: pkg/addons/default/assets/*
 	env GOBIN=$(GOBIN) time go generate ./$(@D)
 
-pkg	/nodebootstrap/assets.go: pkg/nodebootstrap/assets/*
```
Comment (errordeveloper) [inline Makefile]: I believe this was fixed in master already.

**Comment by errordeveloper** [comment]
@davidhao3300 could you please rebase this (please don't use the 'Update Branch' button, as it introduces merge commits). The change looks good to me.

**Comment by gemagomez** [comment]
Can we add a test case for this?

**Comment by errordeveloper** [comment]
I think adding something based on what we have for AL2 should be sufficient.

Here is what I'd copy:
https://github.com/weaveworks/eksctl/blob/86e1248edb6ea9cdc0253912c11ae76fa267fcef/pkg/cfn/builder/api_test.go#L1618-L1629


And add it one of the Ubuntu cases:

https://github.com/weaveworks/eksctl/blob/86e1248edb6ea9cdc0253912c11ae76fa267fcef/pkg/cfn/builder/api_test.go#L1824-L1867

File: Makefile (hunk 1 above)
Comment (davidhao3300) [reply Makefile]: doesn't seem like it

**Comment by davidhao3300** [comment]
Good to go!

**Comment by davidhao3300** [comment]
daily bump

**Comment by davidhao3300** [comment]
good to merge?

**Comment by errordeveloper** [comment]
Thanks @davidhao3300!

**Comment by davidhao3300** [comment]
Thank you!

**Pull request merged**